src/types.ts

```diff
@@ -1,5 +1,75 @@
-import { FunctionInput } from "@azure/functions";
+import {
+    FunctionInput,
+    FunctionHandler,
+    InvocationContext,
+    FunctionResult,
+    FunctionOutput,
+    FunctionTrigger,
+} from "@azure/functions";
+import { DurableOrchestrationClient, EntityState } from "../src/classes";
+import { DurableEntityBindingInfo } from "../src/durableentitybindinginfo";
+import { IEntityFunctionContext } from "../src/ientityfunctioncontext";
+import { IOrchestrationFunctionContext } from "../src/iorchestrationfunctioncontext";
+import { OrchestratorState } from "../src/orchestratorstate";
+import { DurableOrchestrationInput } from "../src/testingUtils";
 
+// orchestrations
+export type OrchestrationHandler = (
+    context: IOrchestrationFunctionContext
+) => Generator<unknown, unknown, any>;
```

Comment:
What do you mean?

Reply:
Can we use types better than `unknown` and `any`? We should avoid those whenever possible - it means you get zero intellisense/build errors/etc.

Reply:
I will actually defer to @davidmrdavid on this one. This type signature for the genrator `Genrator<unknown, unknown, any>` predates me, I just gave it a name. I don't think I know enough about generators or their type constraints to answer that question. @davidmrdavid, do you know if we can enforce stricter types here?

Reply:
I've been trying to answer this question myself for a while, but have never had enough JS/TS experts around me to give it a serious shot. Let's try it now...

So the signature of a Generator is `Generator<InputType, ReturnType, YieldType>`.

Some definitions
*  The input type I believe stands for the inputs that you give the generator every time you ask it to run up to the next yield or the return statement. 
* The return type is the type of data at the return statement
* The YieldType is the type of intermediate values

So I think we may be able to constrain `YieldType` to be of type "Task" (concretely, it may be `DFTask`, can't remember at this time). Users are only allowed to yield `Task` types, and if they're not yield'ing `Task`s then they should encounter an exception. So this is a type we could try to constraint.

For the `ReturnType`, that's hard. Orchestrators may return anything, but they may return only once. Is it possible to parametrize `OrchestrationHandler` with a generic parameter `T` that corresponds to the return type? If so, then yes: we could rename `ReturnType` with that generic `T` so that it's fixed.

For `InputType` - I don't think we can constraint that one at all. The orchestration generator is given as **inputs** the results of each DF API that is yielded. Those DF APIs may truly return anything.

@ejizba/@hossam-nasr: thoughts?

Reply:
I don't have a quick answer, and also I'm not very familiar with Generators. For this PR I'm fine sticking with the status quo since Hossam said it predates his changes